tests/www/test_app.py

```diff
@@ -215,3 +217,27 @@ def test_should_set_sqlalchemy_engine_options(self):
             'max_overflow': 5
         }
         self.assertEqual(app.config['SQLALCHEMY_ENGINE_OPTIONS'], engine_params)
+
+    @conf_vars({
+        ('webserver', 'session_lifetime_minutes'): '3600',
+    })
+    @mock.patch("airflow.www.app.app", None)
+    def test_should_set_permanent_session_timeout(self):
+        app = application.cached_app(testing=True)
+        self.assertEqual(app.config['PERMANENT_SESSION_LIFETIME'], timedelta(minutes=3600))
+
+    @conf_vars({
+        ('webserver', 'session_lifetime_days'): '30',
+        ('webserver', 'force_log_out_after'): '30',
+    })
+    @mock.patch("airflow.www.app.app", None)
+    def test_should_emit_deprecation_warnings(self):
+        with pytest.warns(DeprecationWarning) as warns:
+            application.cached_app(testing=True)
+        app_path = str(Path(__file__).parents[2].joinpath('airflow', 'www', 'app.py'))
+        warns = [w for w in warns if w.filename == app_path]
+        self.assertEqual(len(warns), 2)
+        self.assertEqual(warns[0].message.args[0],
```

Comment:
My main issue with this function is that if we tomorrow add new deprecation warnings it can break this test, while it shouldn't.
Why not just check in general that those messages exists in warns? Why filter by filename and be specific on the index number?